### PR TITLE
EAD/EAD-Session2 assignments

### PR DESCRIPTION
**Pull request by meta-ramjilal-choudhary**

### CODE REVIEW

**Comment by meta-ramjilal-choudhary** [comment]
@metaamitasharma

File: EAD/EAD-Session2/sign_up.html [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+                                <option> Metacube </option>
+                                <option> Google </option>
+                                <option> Microsoft </option>
+                                <option> Rubric Inc. </option>
+                                <option> J P Morgan </option>
+                            </select>
+                        </td>
+                    </tr>
+                    <tr>
+                        <td colspan = "2" align="center"> <input type="button" name="Submit" value="Submit" onclick="validation()" /> </td>
```
Comment (metaamitasharma) [inline EAD/EAD-Session2/sign_up.html]: validateInfo or validateForm

File: EAD/EAD-Session2/validation.js [hunk 2]
```diff
@@ -0,0 +1,76 @@
+function nameValidation(){
```
Comment (metaamitasharma) [inline EAD/EAD-Session2/validation.js]: if your method is  returning boolean whether name is valid or not give the method name validateName or isNameValid. Applicable for all method names.

File: EAD/EAD-Session2/validation.js [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+    var first_name = document.getElementById("first_name");
```
Comment (metaamitasharma) [inline EAD/EAD-Session2/validation.js]: get value here as variable name is more significant for value not for that element.

File: EAD/EAD-Session2/validation.js [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+    if(first_name.value.length <= 2 || last_name.value.length <= 2){
+        debugger;
+        alert("Please insert a valid name");
+        return false;
+    }
+    if(first_name.value.match(latters || last_name.value.match(latters))){
+        return true;
+    }
+    else{
+        alert("Please insert a valid name");
```
Comment (metaamitasharma) [inline EAD/EAD-Session2/validation.js]: if you are showing same message then why two if ? better is to throw exact message that what is going wrong.

File: EAD/EAD-Session2/validation.js [hunk 5]
```diff
@@ -0,0 +1,76 @@
+function nameValidation(){
+    var first_name = document.getElementById("first_name");
+    var last_name = document.getElementById("last_name");
+    var latters = /^[A-Za-z]+$/;
+    if(first_name.value.length <= 2 || last_name.value.length <= 2){
```
Comment (metaamitasharma) [inline EAD/EAD-Session2/validation.js]: optimize

File: EAD/EAD-Session2/validation.js [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+        alert("Please insert a valid Email");
+        return false;
+    }
+}
+
+function passwordValidation(){
+    var pwd = document.getElementById("password");
+    var confirm_pwd = document.getElementById("confirm_pwd");
+    var validPwdPattern = /^(?=.*\d)(?=.*[a-z])(?=.*[A-Z])(?=.*[^a-zA-Z0-9])(?!.*\s).{8,15}$/;
+    if(pwd.value.match(validPwdPattern) && confirm_pwd.value.match(validPwdPattern)){
```
Comment (metaamitasharma) [inline EAD/EAD-Session2/validation.js]: check the password pattern and match it with confirm_pwd is sufficient I think.

File: EAD/EAD-Session2/validation.js [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+    if(email.value.match(validEmailPattern)){
+        return true;
+    }
+    else{
+        alert("Please insert a valid Email");
+        return false;
+    }
+}
+
+function passwordValidation(){
```
Comment (metaamitasharma) [inline EAD/EAD-Session2/validation.js]: optimize

File: EAD/EAD-Session2/validation.js [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+        return true;
+    }
+    else{
+        alert("Please insert a valid Contact number");
+        return false;
+    }
+}
+
+function validation(){
+    if(nameValidation() == false){
```
Comment (metaamitasharma) [inline EAD/EAD-Session2/validation.js]: if(! validateName())

File: EAD/EAD-Session2/validation.js [hunk 9]
```diff
@@ -0,0 +1,76 @@
...
+    if(contact.value.match(contactValidPattern)){
+        return true;
+    }
+    else{
+        alert("Please insert a valid Contact number");
+        return false;
+    }
+}
+
+function validation(){
```
Comment (metaamitasharma) [inline EAD/EAD-Session2/validation.js]: optimize

File: EAD/EAD-Session2/log_in.html [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+                    <tr>
+                        <td>Email</td>
+                        <td> <input type="text" name="name"> </td>
+                    </tr>
+                    <tr>
+                        <td> Password </td>
+                        <td> <input type="password" name="password"/> </td>
+                    </tr>
+                    <tr>
+                        <td colspan="2" align="center"> <input type="submit" name="submit" value="Log in" /> </td>
```
Comment (metaamitasharma) [inline EAD/EAD-Session2/log_in.html]: missing validations.

File: EAD/EAD-Session2/profile.html [hunk 11]
```diff
@@ -0,0 +1,60 @@
...
+    </head>
+    <body>
+        <div class="head">
+            <table width="100%">
+                <tr>
+                    <td align="center">
+                        <img src="html_logo.png" alt="HTML_LOGO" height="30"> 
+                    </td>
+                    <td align="right">
+                        <a href="#"> Home </a>
```
Comment (metaamitasharma) [inline EAD/EAD-Session2/profile.html]: only username and logout.

File: EAD/EAD-Session2/profile.html [hunk 12]
```diff
@@ -0,0 +1,60 @@
...
+                    <tr>
+                        <th align="left"> Contact Number</th>
+                        <td> 8385000000 </td>
+                    </tr>
+                    <tr>
+                        <th align="left"> Company </th>
+                        <td> Metacube </td>
+                    </tr>
+                    <tr>
+                        <td colspan="2" align="center"> <input type="submit" name="submit" value="Edit Field" /> </td>
```
Comment (metaamitasharma) [inline EAD/EAD-Session2/profile.html]: Button name 'Edit Info'.

File: EAD/EAD-Session2/edit.html [hunk 13]
```diff
@@ -0,0 +1,77 @@
+<!DOCTYPE html>
```
Comment (metaamitasharma) [inline EAD/EAD-Session2/edit.html]: Is this expected as per question ?

**Pull request merged**